server/src/models/Domain.js

```diff
@@ -0,0 +1,27 @@
+const mongoose = require('mongoose');
+
+const domain = new mongoose.Schema({
+    domainName: {
+        type: String,
+        required: true
+    },
+    domainDescription: {
+        type: String,
+        required: true
+    },
+    imageUrl: {
+        type: String,
+        required: true
+    },
+    projects: {
+        type: [{
+          type: mongoose.Schema.Types.ObjectId,
+          ref: 'project'
+        }],
+        required: true
```

Comment:
If domains and projects both are required at the same time, then we will not be able to insert data first from the API. I think required can be false here.

Reply:
I'll set `required` to `false` and `default` to an empty array